content/information-distribution/source.md

```diff
@@ -75,6 +77,63 @@ In such an event, client implementations **should** consider articles posted
 prior to the key being declared as compromised as probably not having been
 tampered with, with no way to be 100% sure about it.
 
+## Change of Matrix user
+
+A source or sub-source might have to change the Matrix user it uses to publish
+information at some points in its life time. This can happen for many reason:
+the homeserver it was registered on went down, or got compromised, or got
+isolated from the rest of the federation...
+
+In such an event, the source or sub-source **must** publish a new registration
+event from its new Matrix user with the `prev_id` key having its previous user's
+ID as its value, and the `prev_event` key having the ID of the latest event
+emitted prior to the change as its value.
+
+This key **must** be used by client implementations to link the previous user to
+the new one, considering both as the same entity. A missing `prev_id` key or an
+empty string simply means that the source or sub-source didn't previously use
+another Matrix account to publish information. In this case, the `prev_event`
+key can be omitted as well.
+
+If the entity is a source (and not a sub-source), trust authorities certifying
+it as trustworthy **must** update their list of trusted entities by removing the
+previous user from it, and adding the new one to it. The signature associated
+with the new user **must** be generated from the new user's registration event.
+
+Client implementations **should** define a threshold for linking a source to its
+new user. This threshold is defined by the percentage of trust authorities
+certifying this source as trustworthy that updated their list of trusted sources
+to reference the new user, within the trust authorities trusted by the user. If
+a threshold is defined by a client implementation, it **should** consider the
+link between a source and its new user as valid once that percentage becomes
+higher than the threshold. Such a threshold isn't necessary for sub-sources,
+because a sub-source is only verified by its parent source, which then **must**
+update its list of sub-sources by removing the sub-source's previous user and
+adding the new one.
+
+Once the link between a source (or a sub-source) and its new user is consider
+valid, client implementations **must** consider both the source's (or
+sub-source's) new user and its previous one (and older ones if the source or
+sub-source changed its user more than once) as the same entity. This means that
+articles published by the source's (or the sub-source's) previous user **must**
+be treated as if it was published by the new user, and a reference to the
+source's previous user in a trust authority's list of trustworthy sources
+**must** be considered as a reference to the new user (with the exception of the
+cryptographic check on the source's registration, which is still done using the
+registration event published by the previous user).
```

Comment:
> a reference to the source's previous user in a trust authority's list of trustworthy sources **must** be considered as a reference to the new user

I don't think it's a good idea to alter the TA trusted entities list client-side:
- That would make TA appear as if they trust sources they didn't explicitly trust. That means if a client makes a wrong decision when validating the new user change, the change would then appear as 100% accepted (which is probably not the case).
- Verifying the new source's registration event against the old signature is going to fail in some cases (every time for localized sources with subsources list)

Reply:
ftr I let the "how the client implementation considers a link valid" (or "accepted" as you name it) unspecified on purpose to let implementations manage that the way they see fit. What is being specified here is what's happening once the client implementation decides the link is valid, either by using a threshold or having the user validate it. Either way, there's a point from which a client implementation will be able to state "now I'm pretty sure this new user is the source it pretends to be", and this paragraph specifies the UX it needs to follow from that point.

I apparently need to add some more details, as it's indeed not clear on how that impact crypto checks (which it shouldn't, the way I see it once the link is validated the "source" entity is defined by both MXIDs, and crypto checks should succeed as long as the MXID used is either the new one of an old one, and in the latter case as long as its registration has been emitted before `last_event`).

Reply:
Re-reading the paragraph, it already says:

> a reference to the source's previous user in a
> trust authority's list of trustworthy sources **must** be considered as a
> reference to the new user (with the exception of the cryptographic check on the
> source's registration, which is still done using the registration event
> published by the previous user).

which is pretty much what I said in my previous comment. I'll complete it to mention the cryptographic check is still done using the previous MXID, and might add a short sentence making clear that the source, as an entity, is defined by all of the MXIDs it's used, but otherwise there's not much to add.